src/services/catalog.service.ts

```diff
@@ -35,8 +38,46 @@ export class CatalogService {
         }
       });  // .ea category
     });  // .ea tag
+    if (_.isEmpty(catsBySubCats)) {
+      catsBySubCats[otherId] = otherId;
+    }
     return catsBySubCats;
   }
+
+  public hasCategory(item: any, category: string) {
+    return _.includes(item.catsBySubCats, category);
+  }
+
+  public hasSubCategory(item: any, subCategory: string) {
+    return _.includes(Object.keys(item.catsBySubCats), subCategory);
+  }
+
+  /**
+   * Return a new Array of only those Categories and SubCategories which
+   * exist in the passed in items.
+   * @param items
+   * @returns {Array}
+   */
+  public removeEmptyCategories(items: IServiceItem) {
```

Comment:
This might be difficult currently since the controller calculates rows and columns, but I don't think we should need to remove the empty categories from the model. Instead it's probably easier just to have an ng-if not empty check in the ng-repeat in the view.

Not in this PR, but as a follow-on, I'd like to look at how we're handling this generally. I think we need to get rid of the rows and columns to make the view responsive, and it might make things easier for search as well.